v0/destinations/personalize/transform.js

```diff
@@ -22,79 +16,64 @@ async function process(event) {
   if (message.event === event.destination.Config.eventName) {
     for (let k = 0; k < noOfFields; k += 1) {
       if (
-        keyField[k].toUpperCase() === "USER_ID" ||
-        keyField[k].toUpperCase() === "EVENT_TYPE" ||
-        keyField[k].toUpperCase() === "TIMESTAMP"
+        !(
+          keyField[k].toUpperCase() === "USER_ID" ||
+          keyField[k].toUpperCase() === "EVENT_TYPE" ||
+          keyField[k].toUpperCase() === "TIMESTAMP"
```

Comment:
add item_id here and don't need to delete later.

Reply:
same for the other fields you are deleting

Reply:
remove the condition for timestamp here. check later that if timestamp is not present in this property (ie user did not add timestamp field in the web app) then assign it to getFieldValueFromMessage(historicalTimestamp) and convert to epoch